documentation/specs/dotnet-run-for-maui.md

```diff
@@ -0,0 +1,209 @@
+# `dotnet run` for .NET MAUI Scenarios
+
+The current state of `dotnet run` for .NET MAUI projects is summarized
+by this issue from 2021:
+
+* [Figure out 'dotnet run' CLI experience for iOS and Android](https://github.com/dotnet/xamarin/issues/26)
+
+The pain points being:
+
+* iOS and Android use different properties to select a device,
+  emulator, or simulator. It is difficult for developers to make this
+  selection as you need to run platform-specific commands along with
+  complex MSBuild properties.
+
+* Each platform has different behavior, regards to displaying console
+  output, etc.
+
+* Using an MSIX with WindowsAppSDK doesn't support `dotnet run` at all,
+  relying completely on IDEs like Visual Studio.
+
+* `dotnet run` does not have a concept of a "deploy" step.
+
+This has become more relevant in the AI era, as someone is going to
+expect AIs in "agent mode" to build and run their app. If the
+command-line options are difficult, AIs will fail just as badly as
+humans are today.
+
+## The `dotnet run` Pipeline
+
+These are the high-level steps during `dotnet run`, that we would like
+to make extensible for .NET MAUI (and future) scenarios.
+
+* `restore`: unchanged
+
+* "Pre-run evaluation"
+
+  * If the project is multi-targeted, containing the
+    `$(TargetFrameworks)` property, and `-f` was not supplied...
```

Comment:
```suggestion
    `$(TargetFrameworks)` property _and_ that property has more than one item in it, and `-f` was not supplied...
```

Reply:
We can add a nice UX feature pretty easily here since there are nonzero folks (especially internally) who use TargetFrameworks even for a single TFM to make it easier to add/remove platforms.